main.nf

```diff
@@ -48,6 +48,43 @@ process SYNAPSE_GET {
     """
 }
 
+// calculates MD5 Checksum from contents of downloaded Synapse file and checks it against
+// the one provided in the input CSV file
+process MD5_VALIDATE {
+
+    container "python:3.10.4"
+
+    input:
+    tuple val(syn_id), val(type), val(version), val(md5_checksum), path(path)
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** Given the number of metadata values you are passing between processing, I think we should consider adopting the nf-core approach. Briefly, they "package" up the metadata values into one hash map (similar to Python dictionaries) called `meta`. 

Here's an [example](https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/modules/nf-core/modules/star/align/main.nf#L11) of it being used in the `nf-core/rnaseq` pipeline, and here's [how they prepare it](https://github.com/nf-core/rnaseq/blob/e0dfce9af5c2299bcc2b8a74b6559ce055965455/subworkflows/local/input_check.nf). In our case, we just need to delete the following `map {}` operation in the `workflow` block and update our processes to expect `val(meta)` as their first input. 

```nextflow
        | map {tuple(it.synapse_id, it.type, it.version_number, it.md5_checksum)} \
```

Then, in the `script:` block, you refer to `meta.syn_id`, etc. Admittedly, this will lengthen the commands, but I think it's worth simplifying the inputs/outputs of each process, IMHO.

Reply:
I agree with this Nit.